libminifi/src/core/Processor.cpp

```diff
@@ -386,6 +379,39 @@ std::shared_ptr<Connectable> Processor::pickIncomingConnection() {
   return getNextIncomingConnectionImpl(rel_guard);
 }
 
+using annotation::input::EInputRequirement;
+
+void Processor::validateAnnotations() const {
+  switch (getInputRequirement()) {
+    case EInputRequirement::INPUT_ALLOWED:
+      return;
+    case EInputRequirement::INPUT_REQUIRED: {
+      if (!hasIncomingConnections()) {
+        throw Exception(PROCESS_SCHEDULE_EXCEPTION, "INPUT_REQUIRED was specified for the processor, but no inputs were found");
+      }
+      break;
+    }
+    case EInputRequirement::INPUT_FORBIDDEN: {
+      if (hasIncomingConnections()) {
+        throw Exception(PROCESS_SCHEDULE_EXCEPTION, "INPUT_FORBIDDEN was specified for the processor, but at least one input was found");
+      }
+      break;
+    }
+  }
+}
+
+EInputRequirement Processor::getInputRequirement() const {
+  // default input requirement
+  EInputRequirement myInputRequirement = EInputRequirement::INPUT_ALLOWED;
+  // get own input annotation if InputRequirementAnnotationBase is a base class
```

Comment:
this comment got out of sync with the removal of `InputRequirementAnnotationBase`

Reply:
Done.